IdentityCore/src/webview/embeddedWebview/challangeHandlers/mac/MSIDCertAuthHandler.m

```diff
@@ -72,7 +76,16 @@ + (BOOL)handleChallenge:(NSURLAuthenticationChallenge *)challenge
                  completionHandler(NSURLSessionAuthChallengeRejectProtectionSpace, nil);
                  return;
              }
-             
+            
+            // If there is no preferred identity saved, we must set preferred identity certificate using hostname and key usage parameters
+            if (identity == NULL)
+            {
+                NSArray *arr = @[(__bridge NSString *)kSecAttrCanSign, (__bridge NSString *)kSecAttrCanEncrypt, (__bridge NSString *)kSecAttrCanDecrypt];
+                CFArrayRef arrayRef = (__bridge CFArrayRef)arr;
```

Comment:
Would you check if you need to call ```CFReleaseNull``` to release ```arrayRef```?

Reply:
Sure thing, I tried releasing `arrayRef` after it's finished being used, but the app seems to crash at that. I receive a `EXC_BAD_ACCESS (code=1)...` error so I'm assuming the object is already released? Let me know if I'm understanding incorrectly.

Reply:
Sure thing, I tried releasing `arrayRef` after it's finished being used, but the app seems to crash at that. I receive a `EXC_BAD_ACCESS (code=1)...` error so I'm assuming the object is already released? Let me know if I'm understanding incorrectly. Currently leaving code as is, without `CFRelease`